### ISSUE TITLE
Eager-loading

### ISSUE THREAD

**Issue by Seb-C** [Seb-C/kiss-orm#6]
During a discussion on HN I had an interesting idea on how to implement a basic but useful eager-loading in kiss-orm.

I want to see what can be achieved.

https://news.ycombinator.com/item?id=24371160

**Comment by Seb-C** [comment]
To achieve that, I want to introduce a breaking change and rework completely the design of the relationships.

Instead of declaring the relationships in the repository constructor, you would have to manually define a method for every relationship.

For example, instead of this:
```typescript
class RoleRepository extends CrudRepository<RoleModel> {
    constructor(database: PgSqlDatabase) {
        super({
            // [...]
            relationships: {
                users: (role: RoleModel) => (new UserRepository(database)).search(sql`"roleId" = ${role.id}`),
            },
        });
    }
}
```

The new design would be something like that:
```typescript
class RoleRepository extends CrudRepository<RoleModel> {
    constructor(database: PgSqlDatabase) {
        super({
            // [...]
        });
    }

    public async loadUsersRelationship(role: Role) {
        role.users = await (new UserRepository(this.database)).search(sql`"roleId" = ${role.id}`);
    }
}
```

This new design is way more clean and would allow me to introduce simple protected methods to easily define eager loading methods.

**Comment by Seb-C** [comment]
So I tried to research a different design for eager-loading.
But anything I could think about ends-up being too complex to abstract.

The best I could get was this [design](https://github.com/Seb-C/kiss-orm/blob/feature/eagerLoading/README.md#eager-loading-for-relationships). However it is not usable for many-to-many relationships, the added method does not really feel like a useful addition and it adds complexities to handle like the empty `IN()` case.

So I decided to stick to my original design and still not support eager-loading.
